Core/MethodReturn/MethodReturnNullAnnotator.cs

```diff
@@ -31,20 +31,25 @@ public MethodReturnNullAnnotator (SemanticModel semanticModel)
 
     public override SyntaxNode? VisitMethodDeclaration (MethodDeclarationSyntax node)
     {
-      if (NullUtilities.DoesReturnVoid (node))
-        return node;
-      if (node.Body == null)
-        return node;
-      if (node.Body?.Statements.Count == 0)
+      if (NullUtilities.ReturnsVoid (node)
+          || HasNullOrEmptyBody (node))
+      {
         return node;
+      }
 
-      var returnsNull = HasCanBeNullAttribute (node)
-                        || NullUtilities.ReturnsNull (node, _semanticModel);
+      if (HasCanBeNullAttribute (node)
+          || NullUtilities.ReturnsNull (node, _semanticModel))
+      {
+        return NullUtilities.ToNullReturning (node);
+      }
 
-      if (!returnsNull)
-        return node;
+      return node;

```

Comment:
Could this be simplified to 2 possible paths instead of 3? I'm thinking 1 `if` that returns converted `node` and a "default" case returning the untouched `node`.